seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/source/ElasticsearchSourceReader.java

```diff
@@ -74,22 +79,11 @@ public void pollNext(Collector<SeaTunnelRow> output) throws Exception {
             ElasticsearchSourceSplit split = splits.poll();
             if (split != null) {
                 SeaTunnelRowType seaTunnelRowType = split.getSeaTunnelRowType();
-                SeaTunnelRowDeserializer deserializer =
-                        new DefaultSeaTunnelRowDeserializer(seaTunnelRowType);
                 ElasticsearchConfig sourceIndexInfo = split.getElasticsearchConfig();
-                ScrollResult scrollResult =
-                        esRestClient.searchByScroll(
-                                sourceIndexInfo.getIndex(),
-                                sourceIndexInfo.getSource(),
-                                sourceIndexInfo.getQuery(),
-                                sourceIndexInfo.getScrollTime(),
-                                sourceIndexInfo.getScrollSize());
-                outputFromScrollResult(scrollResult, sourceIndexInfo, output, deserializer);
-                while (scrollResult.getDocs() != null && scrollResult.getDocs().size() > 0) {
-                    scrollResult =
-                            esRestClient.searchWithScrollId(
-                                    scrollResult.getScrollId(), sourceIndexInfo.getScrollTime());
-                    outputFromScrollResult(scrollResult, sourceIndexInfo, output, deserializer);
+                if (SQL_TYPE.equalsIgnoreCase(sourceIndexInfo.getSearchType())) {
+                    cursorSearchResult(seaTunnelRowType, sourceIndexInfo, output);
+                } else {
+                    scrollSearchResult(seaTunnelRowType, sourceIndexInfo, output);
```

Comment:
what's different between `cursorSearchResult` and `scrollSearchResult`? Can we merge it?

Reply:
@CosmosNi